compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -4677,7 +4677,13 @@ private void checkTypesForRecords(BLangWaitForAllExpr waitExpr, AnalyzerData dat
 
         for (BLangWaitForAllExpr.BLangWaitKeyValue keyVal : rhsFields) {
             String key = keyVal.key.value;
-            if (!lhsFields.containsKey(key)) {
+            BLangExpression valueExpr = keyVal.valueExpr;
+            if (valueExpr != null && valueExpr.getKind() == NodeKind.BINARY_EXPR
```

Comment:
Can the valueexpr be nested inside a grouped-expr?

Reply:
According to the spec, https://ballerina.io/spec/lang/master/#section_7.7.2
It can be.
Will update the PR with necessary changes

Reply:
Updated. 8e11173274edb73476af9ec35407f2553d8a3c89